library/src/schemas/enum/enum.ts

```diff
@@ -18,6 +18,48 @@ export interface Enum {
   [key: string]: string | number;
 }
 
+/**
+ * Checks if a string can be produced by Number.prototype.toString with radix=10.
+ */
+type IsNumericString<K extends string> = K extends
+  | 'NaN'
+  | 'Infinity'
+  | '-Infinity'
+  ? true
+  : K extends `${infer V extends number}`
+    ? `${V}` extends K
+      ? true
+      : false
+    : false;
+
+type OmitReverseMappingValue<
+  K extends string | number,
+  V,
+  Ks extends string | number,
+> = K extends number
+  ? number extends K
+    ? never
+    : Exclude<V, Ks>
+  : K extends string
+    ? IsNumericString<K> extends true
+      ? Exclude<V, Ks>
+      : V
+    : never;
+
+/**
+ * enum_ only accepts enum values of string & non-numeric keys that has its reverse mapping.
+ * This type-level function filters out numeric keys including Infinity and NaN.
+ *
+ * @example EnumValues<{ 1: NaN; NaN: 1; 2: 'foo' }> = NaN | 'foo'
+ */
+type EnumValues<TEnum extends Enum> = {
```

Comment:
I would change the implementation of `EnumValues`. I think the current code produces wrong types in some cases. I tried to "copy" the JS code we use for filtering to the TS type.

```ts
type EnumValues<TEnum extends Enum> = {
  [TKey in keyof TEnum]: TKey extends number
    ? TEnum[TKey] extends string
      ? TEnum[TEnum[TKey]] extends TKey
        ? never
        : TEnum[TKey]
      : TEnum[TKey]
    : TKey extends 'NaN' | 'Infinity' | '-Infinity'
      ? TEnum[TKey] extends string
        ? TEnum[TEnum[TKey]] extends number
          ? never
          : TEnum[TKey]
        : TEnum[TKey]
      : TKey extends `${infer TNumber extends number}`
        ? TEnum[TKey] extends string
          ? TEnum[TEnum[TKey]] extends TNumber
            ? never
            : TEnum[TKey]
          : TEnum[TKey]
        : TEnum[TKey];
}[keyof TEnum];
```

Reply:
Once we have decided how to proceed with `EnumValues`, I will do a final review and merge this PR.